### PR TITLE
Add study publications to SCP API (SCP-5595)

### PR DESCRIPTION
**Pull request by eweitz**
This integrates studies' journal publications into SCP API, enabling better programmatic accessibility.

### Overview
[Previously](#1379), study editors and owners could add their publication's structured data, and it conveniently displayed at right in the Summary tab of the Study Overview page.  However, the data was only accessible via the UI, and not in a RESTful format.

Now publication data is available via two sets of [SCP API](https://singlecell.broadinstitute.org/single_cell/api/v1) endpoints: `/site/studies/{accession}` and a new group of CRUD endpoints anchored around `/studies/{accession}/publications`.  Implementation reuses patterns from "external resources".  These changes help decouple the frontend and backend, and enable more flexible ways to process publication data.  That promotes FAIR data.

### Limitations
There's a lot that could be improved about publications on SCP!  Some studies link to their publications in the study description, some do via "external resources"; such cases could likely be inferred and better structured in a publication object.  The sidebar UI display for publications could be refined.  Capturing DOI specifically might be nice.  "Citation" values could probably usually be automatically populated.  

This PR only intends to make existing structured data for publications accessible by API.

### Test
Automated tests verify expected behavior.  If you'd like to manually confirm:
1.  Go to a study, e.g. your local "Cellular and transcriptional diversity over the course of human lactation"
2.  Settings -> Authors/Publications
3. Add structured data for publication, e.g.:
- Title: Cellular and transcriptional diversity over the course of human lactation
- Journal: PNAS
- Citation: https://www.pnas.org/doi/10.1073/pnas.2121720119
- URL: https://doi.org/10.1073/pnas.2121720119
4. Click "Update settings"
5. Note your study's accession
6. Go to https://localhost:3000/single_cell/api/v1/site/studies/(your-study-accession)
7. Confirm a `publications` field exists in the response with the data you supplied

This satisfies SCP-5595, a Developer's Choice Days project.

### CODE REVIEW

**Review by jlchang — APPROVED** [review]
Love this step towards FAIR data! Publication object accessible by API as described.

File: app/controllers/api/v1/publications_controller.rb [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+        rescue => e
+          ErrorTracker.report_exception(e, current_api_user, @publication, params)
+          MetricsService.report_error(e, request, current_api_user, @study)
+          render json: {error: e.message}, status: 500
+        end
+      end
+
+      private
+
+      def set_study
```
Comment (bistline) [inline app/controllers/api/v1/publications_controller.rb:284]: We have a concern called `StudyAware` that is included in all `ApiBaseController` descendants.  This provides built-in methods for `:set_study`, `:check_study_view_permission`, and `check_study_edit_permission`.  As this concern is already included in this controller, you can remove `:set_study` and `: check_study_permission` here and just rely on the `before_action` filters you've set up (though you'll need to rename `:check_study_permission` to `:check_study_edit_permission`.

Again, this seems to be a side effect of copying from `ExternalResoucesController`.  You can make the same edits there as well.  That controller predated the `StudyAware` module.

**Review by bistline — CHANGES_REQUESTED** [review]
Great addition, and love that we're still being FAIR.  I note two code simplification edits that will reduce the amount of redundancy and improve latency by removing irrelevant status checks.

Also, unrelated - I initially wasn't able to add the publication because I ran into the "Study files is invalid" [bug](SCP-4142).  This may be worth addressing, either by improving the error message or (even better) avoid re-validating study file entries when all we're trying to do is update study settings.

File: app/controllers/api/v1/publications_controller.rb [hunk 2]
```diff
@@ -0,0 +1,314 @@
+module Api
+  module V1
+    class PublicationsController < ApiBaseController
+      include Concerns::FireCloudStatus
```
[the remark this answers is not in the document]
Comment (eweitz) [reply app/controllers/api/v1/publications_controller.rb]: Done in 62d0940.

File: app/controllers/api/v1/publications_controller.rb (hunk 1 above)
Comment (eweitz) [reply app/controllers/api/v1/publications_controller.rb]: Helpful!  Done in 254af54.  I also widened accessibility for public studies in 02b4b1b.

**Comment by jlchang** [comment]
Functional review - confirmed that manual test still succeeds after latest refinements.

**Review by bistline — APPROVED** [review]

**Pull request merged**